AdyenDropIn/Utilities/ComponentManager.swift

```diff
@@ -117,38 +117,25 @@ internal final class ComponentManager {
     
     private func createApplePayComponent(with paymentMethod: ApplePayPaymentMethod) -> PaymentComponent? {
         guard
-            let summaryItems = configuration.applePay.summaryItems,
-            let identfier = configuration.applePay.merchantIdentifier,
-            let payment = payment else {
+            let payment = payment,
+            let configuration = configuration.applePay.componentConfiguration()
+        else {
+            adyenPrint("Failed to instantiate ApplePayComponent because ApplePayConfiguration is missing data")
             return nil
         }
-        
-        let requiredBillingContactFields = configuration.applePay.requiredBillingContactFields
-        let requiredShippingContactFields = configuration.applePay.requiredShippingContactFields
-        let excludedCardNetworks = configuration.applePay.excludedCardNetworks
-        
-        do {
-            let configuration = ApplePayComponent.Configuration(summaryItems: summaryItems,
-                                                                merchantIdentifier: identfier,
-                                                                requiredBillingContactFields: requiredBillingContactFields,
-                                                                requiredShippingContactFields: requiredShippingContactFields,
-                                                                excludedCardNetworks: excludedCardNetworks)
-            return try ApplePayComponent(paymentMethod: paymentMethod,
-                                         payment: payment,
+
+        return try? PreApplePayComponent(payment: payment,
```

Comment:
you should put this in a `do try`, to print a meaningful message to the console if the instantiation fails.

Reply:
👍